### PR TITLE
Refactor site creation

### PR DESCRIPTION
**Pull request by mauritsvanrees**
This is part of https://github.com/plone/Products.CMFPlone/issues/3961

This affects the site creation process when `plone.distribution` is not available.  See the individual commits for details.  In this case, the plone-overview page on the Zope root looks like this:

Screenshot 2024-06-19 at 20 38 50

The Advanced version of the Add Plone Site form is gone.  For advanced options you should use a Plone Distribution.

This needs a change in `plone.app.testing`, otherwise loads of tests fail.  Only a one line change though.  I will link the PR.

I will keep this as Draft PR for now.

### CODE REVIEW

File: Products/CMFPlone/browser/templates/plone-overview.pt [hunk 1]
```diff
@@ -117,6 +102,11 @@
                 these frontends and possible upgrade paths from older Plone versions
                 to Plone 6.
             </p>
+            <p i18n:translate="help_create_plone_site_buttons_3">
+                Starting with Plone 6.1, you should add the <code>plone.classicui</code>
+                package if you want to create the default content.
+                If you see this text, that means you do not have this package yet.
```
Comment (stevepiercy) [inline Products/CMFPlone/browser/templates/plone-overview.pt:108]: Suggest the following wording.

```
Starting with Plone 6.1, default content is not loaded into the site.
If you want to load the default content, you should install the <code>plone.classicui</code> package.
If you see this text, that means you have not installed that package yet.
```

File: news/3961.breaking.3 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+`factory.addPloneSite`: remove `setup_content` and `content_profile_id` keyword arguments.
+We never setup content anymore.  Use a Plone Distribution if you need this.
```
Comment (stevepiercy) [inline news/3961.breaking.3:2]:
```suggestion
We no longer load default content. Use a Plone Distribution if you need this.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Polish up the messages.

File: Products/CMFPlone/browser/templates/plone-overview.pt (hunk 1 above)
Comment (mauritsvanrees) [reply Products/CMFPlone/browser/templates/plone-overview.pt]: Suggestion applied, thanks.

**Review by stevepiercy — APPROVED** [review]
Thank you! Looks good so far.

**Comment by stevepiercy** [comment]
Upon merge and release, we need to update the documentation. See https://github.com/plone/documentation/issues/1721

**Pull request merged**